aws/resource_aws_acm_certificate.go

```diff
@@ -210,15 +211,16 @@ func resourceAwsAcmCertificateRead(d *schema.ResourceData, meta interface{}) err
 		d.Set("domain_name", resp.Certificate.DomainName)
 		d.Set("arn", resp.Certificate.CertificateArn)
 
-		if err := d.Set("subject_alternative_names", cleanUpSubjectAlternativeNames(resp.Certificate)); err != nil {
+		sans := cleanUpSubjectAlternativeNames(resp.Certificate)
+		if err := d.Set("subject_alternative_names", sans); err != nil {
 			return resource.NonRetryableError(err)
 		}
 
 		domainValidationOptions, emailValidationOptions, err := convertValidationOptions(resp.Certificate)
-
 		if err != nil {
 			return resource.RetryableError(err)
 		}
+		sortDomainValidationOptions(d.Get("domain_name").(*string), sans, domainValidationOptions)
```

Comment:
I tried to use this patch with [terraform-provider-aws v2.23.0](https://github.com/terraform-providers/terraform-provider-aws/commits/v2.23.0), but unfortunately it failed with an error.
I'm not an expert in golang, however it looks to me that `d.Get("domain_name")` returns just `string`, not `*string`.
So changing the type of `domainName` from `*string` to `string` worked for me.